lib/src/builder.rs

```diff
@@ -96,11 +98,40 @@ impl<DB: Database<Error = ProviderError> + DatabaseCommit + OptimisticDatabase>
             _ => unimplemented!(),
         };
 
+        if reth_chain_spec.is_taiko() {
+            let block_num = self.input.taiko.block_proposed.block_number();
+            let block_timestamp = 0u64; // self.input.taiko.block_proposed.block_timestamp();
+            let taiko_fork = self
+                .input
+                .chain_spec
+                .spec_id(block_num, block_timestamp)
+                .unwrap();
+            match taiko_fork {
```

Comment:
Shouldn't reth be able to directly pick the right HF just based on the L2 block number?

Reply:
Yes, the logic is get HF from reth based on l2 number, but as there are 2 chain_spec configs from raiko & reth respectively, code here is to make sure this 2 are the same, i.e., raising error if config mismatch.